pkg/resources/reconciling/tasks_runner.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciling
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TaskID represents the task identifier.
+type TaskID string
+
+// TaskIDSet is a set of TaskIDs
+type TaskIDSet map[TaskID]struct{}
+
+// Has returns true if and only if item is contained in the set.
+func (s TaskIDSet) Has(item TaskID) bool {
+	_, contained := s[item]
+	return contained
+}
+
+// TaskFn represents a reconciliation task.
+type TaskFn func(context.Context, ctrlruntimeclient.Client) error
```

Comment:
```suggestion
type TaskFn func(context.Context) error
```

controller-runtime Client is a leaking abstraction that doesn't belong to the DAG library, we can always pass inside AddTask the wrapped in closure anything we want.

Reply:
I agree with your comment in general, and I thought about this but I came up to the conclusion that the client will be probably needed on every task we will run with this lib, which is not really intended to be general-purpose. This could avoid unneeded wrapping in some cases. I will think about this again. Thx for the hint.